src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -118,7 +114,7 @@ export function getOptionsFromAppInfo( appInfo: AppInfo ): InstanceOptions {
  * @param {InstanceOptions} defaultOptions - options to be used as default values for prompt
  * @returns {any} instance data
  */
-export async function promptForArguments( preselectedOptions: InstanceOptions, defaultOptions: InstanceOptions ): InstanceData {
+export async function promptForArguments( preselectedOptions: InstanceOptions, defaultOptions: $Shape<InstanceOptions> ): Promise<InstanceData> {
```

Comment:
1. `async` functions always return Promises.
2. When we call `promptForArguments()`, we may pass a subset of `InstanceOptions`; therefore, change the type to `$Shape<InstanceOptions>`.